submissions/asaMitaka/FriendsApp/script.js

```diff
@@ -0,0 +1,237 @@
+    try {
+        response = await fetch(url);
+        if (!response.ok) {
+            throw new Error(response.statusText);
+        } 
+    } catch (err) {
+        console.log(err);
+        resetPage();
+    }
+    let jsonData = await response.json();
+
+    friendsCopy = jsonData.results;
+    friends = jsonData.results;
+
+    renderAllItemsToPage(friends);
+}
+
+function creatingDivItem(item) {
+    return `
+        <div class='content__item'>
+            <img class='content__item-img' src='${item.picture.medium}'>
+            <div class='content__item-name'>${item.name.title} ${item.name.first} ${item.name.last}</div>
+            <p class='content__item-email'>
+                <a href="mailto:  ${item.email}"   target="_blank"> ${item.email}</a>
+            </p>
+            <p>Age: ${item.dob.age}</p>
+            <a href='${item.phone}'>${item.phone}</a>
+            <p class='content__item-country'>${item.location.country}, ${item.location.city}</p>
+            <div class='content__item-gender'>${item.gender}</div>
+        </div>
+    `;
+}
+
+form.last.addEventListener('click', function(e) {
+    if (e.target && e.target.classList.contains('last')) {
+        checkLastSort(e);
+    }
+});
+
+function checkLastSort(event) {
+    form.age.reset();
+    switch(event.target.value) {
+        case "lastAToZ":
+            sortedAtoZLast();
+            break;
+        case "lastZToA":
+            sortedZtoALast();
+            break;
+    }
+}
+
+form.reset.reset.addEventListener('click', resetPage);
+
+form.age.addEventListener('click', function(e) {
+    if (e.target && e.target.classList.contains('age')) {
+        sortByAge(e);
+    }
+});
+
+function sortByAge(event) {
+    form.last.reset();
+    switch(event.target.value) {
+        case "ageMore":
+            sortedAgeMore();
+            break;
+        case "ageLess":
+            sortedAgeLess();
+            break;
+    }
+}
+
+function sortedAgeMore() {
+    friends.sort((a, b) => a.dob.age - b.dob.age);
+    renderAllItemsToPage(friends);
+}
+
+function sortedAgeLess() {
+    friends.sort((a, b) => b.dob.age - a.dob.age);
+    renderAllItemsToPage(friends);
+}
+
+function sortedAtoZLast() {
+    friends.sort((a, b) => a.name.last !== b.name.last ? a.name.last < b.name.last ? -1 : 1 : 0);
+    renderAllItemsToPage(friends);
+}
+
+function sortedZtoALast() {
+    friends.sort((a, b) => a.name.last !== b.name.last ? a.name.last > b.name.last ? -1 : 1 : 0);
+    renderAllItemsToPage(friends);
+}
+
+form.gender.addEventListener('click', function(e) {
+    if (e.target && e.target.classList.contains('gender')) {
+        checkedGender(e);
+    }
+});
+
+function checkedGender(event) {
+    switch (event.target.value) {
+        case 'male':
+            renderMale();
+            break;
+        case 'female':
+            renderFemale();
+            break;
+        case 'all':
+            renderAll();
+            break;
+    }
+}
+
+function renderMale() {
```

Comment:
Your `renderMale`, `renderFemale` and `renderAll` functions are very similar. Please combine them.